From 0b8087323f6f66cb03f1c243c312be37b882152f Mon Sep 17 00:00:00 2001
From: Wojciech Sipak <wsipak@antmicro.com>
Date: Thu, 14 Mar 2024 13:05:56 +0100
Subject: [PATCH 2/2] fix configuration of rust-app

The UART-related options are platform specific and are enabled by default anyway.
With CONFIG_LOG=y and CONFIG_LOG_MINIMAL=n there are cases when the build system
fails to genertate any configuration (for example bl625_dvk).
Even if the application can't be built for such platform, we need to make sure that
a configuration file is generated, which is solved by enabling CONFIG_LOG_MINIMAL.

Since 6e977ae2d54c4f5443f752ce3a88f22043dcbf07 in Zephyr,
the rust application needs CONFIG_CPP

Signed-off-by: Wojciech Sipak <wsipak@antmicro.com>
---
 samples/rust-app/prj.conf | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/samples/rust-app/prj.conf b/samples/rust-app/prj.conf
index 4392540..a5c87fe 100644
--- a/samples/rust-app/prj.conf
+++ b/samples/rust-app/prj.conf
@@ -1,9 +1,10 @@
 CONFIG_RUST=y
 CONFIG_RUST_ALLOC_POOL=y
 CONFIG_SERIAL=y
-CONFIG_UART_NATIVE_POSIX=y
-CONFIG_NATIVE_UART_0_ON_OWN_PTY=y
 CONFIG_USERSPACE=y
 CONFIG_QEMU_ICOUNT=n
 CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
 CONFIG_MINIMAL_LIBC=y
+# linking fails without CPP since 6e977ae2d5 in Zephyr
+CONFIG_CPP=y
-- 
2.44.0

